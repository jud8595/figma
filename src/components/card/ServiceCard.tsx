import React from 'react';
import styled from 'styled-components';
import { bodyBook16, heading32, headline16 } from '../../utility/typography';

export interface ServiceCardData {
  smallTitle: string
  title: string
  content: string
}

export const ServiceCard: React.FC<ServiceCardData> = ({smallTitle, title, content}) => {
    return  (
      <CardStyled>
        <SmallTitle>{smallTitle}</SmallTitle>
        <Title>{title}</Title>
        <Content>{content}</Content>
        <ButtonSquare>{'>'}</ButtonSquare>
      </CardStyled>
    )
}

const CardStyled = styled('div')`
  background-color: ${props => props.theme.colors.primary1};
  color: ${props => props.theme.colors.white};
  border-radius: 8px;
  min-height: 344px;
  padding: 32px;
  text-align: left;

  display: flex;
  flex-direction: column;

  &>*:last-child {
    align-self: flex-end;
    margin-top: auto;
  };
`;

const Title = styled('div')`
  ${heading32}
  margin-bottom: 24px;
`;

const SmallTitle = styled('div')`
  ${headline16}
  margin-bottom: 24px;
`;

const Content = styled('div')`
  max-height: 150px;
  overflow: hidden;
  text-overflow: ellipsis;
  margin-bottom: 8px;
  ${bodyBook16}
`;

const ButtonSquare = styled('button')`
  color: ${props => props.theme.colors.white};
  background-color: ${props => props.theme.colors.accent1};
  width: 64px;
  height: 64px;
  border-radius: 8px;
  padding: 16px;
  border-style: none;
  border-width: 0;
`;
