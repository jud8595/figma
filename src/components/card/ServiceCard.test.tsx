import React from 'react';
import { render } from '@testing-library/react';
import { ServiceCard } from './ServiceCard';
import Theme from '../Theme';

test('renders a ServiceCard', () => {
  const { container } = render(
    <Theme>
      <ServiceCard smallTitle='mySmallTitle' title='myTitle' content='myContent' />
    </Theme>
  )
  expect(container).toMatchSnapshot()
});
