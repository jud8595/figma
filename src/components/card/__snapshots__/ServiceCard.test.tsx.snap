// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a ServiceCard 1`] = `
<div>
  <div
    class="sc-bczRLJ cYicyH"
  >
    <div
      class="sc-dkzDqf kIwMib"
    >
      mySmallTitle
    </div>
    <div
      class="sc-gsnTZi iOMiLO"
    >
      myTitle
    </div>
    <div
      class="sc-hKMtZM eMKIEi"
    >
      myContent
    </div>
    <button
      class="sc-eCYdqJ bBePYn"
    >
      &gt;
    </button>
  </div>
</div>
`;
